sql/analyzer/hoist_select_exists.go

```diff
@@ -127,9 +156,17 @@ type hoistSubquery struct {
 	outerFilters []sql.Expression
 }
 
+type aliasCreatorFn func(alias string) string
+
+type fakeNameable struct {
```

Comment:
nit: this could use a `var _ sql.Nameable = (*nil)(fakeNameable)` or similar

Reply:
Excellent point, adding.